FAQ, sql: improve isolation level and FAQ wording (#936)
pingcap/docs: FAQ.md

#### How is TiDB strongly consistent?

> TiDB implements the Repeatable Read isolation in the Snapshot Isolation level. It does so by using the [Raft consensus algorithm](https://raft.github.io/) to ensure consistency among multiple replicas.

Reviewer: Perhaps something like:
```suggestion
TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATABLE-READ` for compatibility with MySQL. Data is redundantly copied between TiKV nodes using the [Raft consensus algorithm](https://raft.github.io/) to ensure recoverability should a node failure occur.
```
The point I am making here is that raft doesn't provide consistency unless there is a failure event.  It is the transactional protocol on top of Raft that provides it.  Raft provides recoverability.

> TiDB implements Snapshot Isolation consistency, which it advertises as `REPEATABLE-READ` for compatibility with MySQL. Data is redundantly copied between TiKV nodes using the [Raft consensus algorithm](https://raft.github.io/) to ensure recoverability should a node failure occur.

Reviewer 2: IMO, SI is a little different from RR.

Reviewer: @siddontang it does not claim that SI is the same as RR.  It says that it _advertises_ it as RR for compatibility.

> At the bottom layer, TiDB uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine.

Reviewer: ```suggestion
At the bottom layer, TiKV uses a model of replication log + State Machine to replicate data. For the write requests, the data is written to a Leader and the Leader then replicates the command to its Followers in the form of log. When the majority of nodes in the cluster receive this log, this log is committed and can be applied into the State Machine.
```

#### What's the maximum tolerance for time synchronization error of PD?

> Theoretically, the smaller of the time synchronization error, the better. PD can tolerate any synchronization error, but a larger error value means a longer period of service stop during the leader change. If the clock goes back during a leader change, the change process will not proceed until it catches up with the previous leader.

Reviewer 3: ```suggestion
Theoretically, the smaller the time synchronization error, the better. PD can tolerate any synchronization error, but a larger error value means a longer period of service stop during the leader change. If the clock goes back during a leader change, the change process will not proceed until it catches up with the previous leader.
```
